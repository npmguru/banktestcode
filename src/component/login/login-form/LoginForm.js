import React, { useState } from "react";

import { Formik, Form, Field, ErrorMessage } from "formik";
import * as Yup from "yup";

import { useDispatch } from 'react-redux';
import { useHistory } from "react-router-dom";

import UserMiddleware from '../../../core/api/login-action'

const LoginSchema = Yup.object().shape({
  email: Yup.string()
    .email("Invalid email address format")
    .required("Email is required"),
  password: Yup.string()
    .min(6, "Password must be 6 characters at minimum")
    .required("Password is required")
});



const LoginForm = () => {  

  const dispatch = useDispatch();
  const history = useHistory();
    return (
        <div className="row">
          <div className="col-lg-12">
            <Formik
              initialValues={{ email: "", password: "" }}
              validationSchema={LoginSchema}
              onSubmit={(userInfo, actions) => {
                actions.setSubmitting(true);
                dispatch(UserMiddleware.userLoginAuth(userInfo, () => { history.push("/dashboard") }));
                setTimeout(function(){ actions.setSubmitting(false) }, 1000)
              }}
            >
              {({ touched, errors, isSubmitting }) => (
                <Form>
                  <div className="form-group">
                    <label htmlFor="email">Email</label>
                    <Field
                      type="email"
                      name="email"
                      placeholder="Enter email"
                      className={`form-control ${
                        touched.email && errors.email ? "is-invalid" : ""
                      }`}
                    />
                    <ErrorMessage
                      component="div"
                      name="email"
                      className="invalid-feedback"
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="password">Password</label>
                    <Field
                      type="password"
                      name="password"
                      placeholder="Enter password"
                      className={`form-control ${
                        touched.password && errors.password ? "is-invalid" : ""
                      }`}
                    />
                    <ErrorMessage
                      component="div"
                      name="password"
                      className="invalid-feedback"
                    />
                  </div>

                  <button
                    type="submit"
                    className="btn btn-primary btn-block"
                    disabled={isSubmitting}
                  >
                    {isSubmitting ? "Please wait..." : "Submit"}
                  </button>
                </Form>
              )}
            </Formik>
          </div>
        </div>
    );
}

export default LoginForm;
