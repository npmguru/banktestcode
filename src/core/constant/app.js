const AUTH_USER = 'AUTH_USER';
const AUTH_SIGNOUT = 'AUTH_SIGNOUT';
const AUTH_ERROR = 'AUTH_ERROR';
const BUSINESS_SERVICE_URI = 'https://secure-dawn-46321.herokuapp.com/api/v1.0';
const ADD_ITEM = 'ADD_ITEM';
const EDIT_ITEM = 'EDIT_ITEM'

export { AUTH_USER, AUTH_ERROR, AUTH_SIGNOUT,BUSINESS_SERVICE_URI,ADD_ITEM,EDIT_ITEM }