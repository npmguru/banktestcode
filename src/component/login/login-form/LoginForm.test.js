import React from 'react';
import Enzyme, { mount } from 'enzyme';
import LoginForm from './LoginForm';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { useDispatch } from 'react-redux';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../../../persistence/reducers';
import { AUTH_USER } from '../../../core/constant/app'
Enzyme.configure({ adapter: new EnzymeAdapter() });


jest.mock("react-redux", () => ({
  connect: () => jest.fn(),
  useSelector: jest.fn(fn => fn()),
  useDispatch: () => jest.fn()
}));

describe('<LoginForm /> with no props', () => {

  afterEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });


  it('should match the snapshot', () => {
    const container = mount(<LoginForm />);
    expect(container.html()).toMatchSnapshot();

  });

  
});