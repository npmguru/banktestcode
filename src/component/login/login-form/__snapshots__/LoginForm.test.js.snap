// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> with no props should match the snapshot 1`] = `"<div class=\\"row\\"><div class=\\"col-lg-12\\"><form action=\\"#\\"><div class=\\"form-group\\"><label for=\\"email\\">Email</label><input name=\\"email\\" type=\\"email\\" placeholder=\\"Enter email\\" class=\\"form-control \\" value=\\"\\"></div><div class=\\"form-group\\"><label for=\\"password\\">Password</label><input name=\\"password\\" type=\\"password\\" placeholder=\\"Enter password\\" class=\\"form-control \\" value=\\"\\"></div><button type=\\"submit\\" class=\\"btn btn-primary btn-block\\">Submit</button></form></div></div>"`;
